BHoM_Adapter_Tests/Create/RandomObjects.cs

```diff
@@ -0,0 +1,93 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Adapter.Tests
+{
+    public static partial class Create
+    {
+        public static List<T> RandomObjects<T>(int count = 100, bool assignIdFragmentWithProgressiveId = false, bool assignObjectName = false, string objectNamePrefix = "bar_") where T : IObject
+        {
+            return RandomBHoMObjects(typeof(T), count, assignIdFragmentWithProgressiveId, assignObjectName, objectNamePrefix).OfType<T>().ToList();
+        }
+
+        public static List<IBHoMObject> RandomBHoMObjects(Type t, int count = 100, bool assignIdFragmentWithProgressiveId = false, bool assignObjectName = false, string objectNamePrefix = "bar_")
+        {
+            List<IBHoMObject> objs = new List<IBHoMObject>();
+
+            for (int i = 0; i < count; i++)
+            {
+                IObject obj = BH.Engine.Base.Create.RandomObject(t);
+
+                if (obj == null)
+                    throw new ArgumentException($"Could not create a Random Object of type {t.FullName}.");
+
+                if (assignIdFragmentWithProgressiveId)
+                {
+                    IBHoMObject bhomObj = obj as IBHoMObject;
+                    TestFragment testIdFragment = new TestFragment() { SomeProp = i };
+                    bhomObj = bhomObj.AddFragment(testIdFragment);
+                    obj = bhomObj;
+                }
+
+                if (assignObjectName)
+                {
+                    IBHoMObject bhomObj = obj as IBHoMObject;
+                    bhomObj.Name = objectNamePrefix + i.ToString();
+                    obj = bhomObj;
+                }
+
+                objs.Add(obj as dynamic);
+            }
+
+            return objs;
+        }
+
+        public static List<IObject> RandomIObjects(Type t, int count = 100)
+        {
+            List<IObject> objs = new List<IObject>();
+
+            for (int i = 0; i < count; i++)
+            {
+                IObject obj = BH.Engine.Base.Create.RandomObject(t);
+
+                objs.Add(obj);
+            }
+
+            return objs;
+        }
+    }
+
+
+    public class TestFragment : IFragment
```

Comment:
👀 this should be its own file

Reply:
87fb96088bc9051e4cd1fed9bbe30728a6a5e519